src/N3Store.js

```diff
@@ -91,26 +91,22 @@ export default class N3Store {
     const varCount = !key0 + !key1 + !key2,
         entityKeys = varCount > 1 ? Object.keys(this._ids) : this._entities;
     const graph = termFromId(graphId, this._factory);
+    const parts = { subject: null, predicate: null, object: null };
```

Comment:
Maybe initialize a constant term as subject/predicate/object so the `parts` object is always uniform?
I suspect the loops and nested ifs might throw off V8 prediction.

E.g., define a random constant term `this._placeholderTerm` as member variable that is created with the factory during construction.

Reply:
> I suspect the loops and nested ifs might throw off V8 prediction.

Could you elaborate on this - since `parts` is never used as a condition or variable in the loops I don't see how it would affect predicition.

Reply:
I mean that V8 loves uniform objects and predictable object evolution, and the parts object will go through 6 different possible shapes (depending on where the nulls are) in different orders. So I wouldn't be surprised if initializing it with Term/Term/Term makes it faster. Then again I started coding against V8 over 10 years ago, and some assumptions (as seen in other PRs) have changed.

Reply:
Ah ok - that being the case why not just have placeholder as a `const PLACEHOLDER_TERM = {}` or `const PLACEHOLDER_TERM = new NamedNode()` at the top of a file rather than as part of the N3 store?

Reply:
I tried 3 approaches and there are no statistically significant difference between them:

Option 1 (current PR):
```
N3Store performance test
- Adding 16777216 triples to the default graph: 12.601s
* Memory usage for triples: 812MB
- Finding all 16777216 triples in the default graph 65536 times (0 variables): 1:26.509 (m:ss.mmm)
- Finding all 16777216 triples in the default graph 131072 times (1 variable): 11.663s
- Finding all 16777216 triples in the default graph 196608 times (2 variables): 17.660s

- Adding 16777216 quads: 20.290s
* Memory usage for quads: 1820MB
- Finding all 16777216 quads 1048576 times: 28.416s
```

Option2
```ts
const parts = {
      subject: this._factory.namedNode(),
      predicate: this._factory.namedNode(),
      object: this._factory.namedNode()
    };
```

```
N3Store performance test
- Adding 16777216 triples to the default graph: 13.252s
* Memory usage for triples: 810MB
- Finding all 16777216 triples in the default graph 65536 times (0 variables): 1:21.621 (m:ss.mmm)
- Finding all 16777216 triples in the default graph 131072 times (1 variable): 11.297s
- Finding all 16777216 triples in the default graph 196608 times (2 variables): 17.893s

- Adding 16777216 quads: 19.041s
* Memory usage for quads: 1806MB
- Finding all 16777216 quads 1048576 times: 29.643s
```

Option3
Globally

```ts
const DEFAULT_NODE = N3DataFactory.namedNode();
```

```ts

    const parts = {
      subject: DEFAULT_NODE,
      predicate: DEFAULT_NODE,
      object: DEFAULT_NODE
    };
```

```
N3Store performance test
- Adding 16777216 triples to the default graph: 12.654s
* Memory usage for triples: 810MB
- Finding all 16777216 triples in the default graph 65536 times (0 variables): 1:20.162 (m:ss.mmm)
- Finding all 16777216 triples in the default graph 131072 times (1 variable): 11.368s
- Finding all 16777216 triples in the default graph 196608 times (2 variables): 19.971s

- Adding 16777216 quads: 21.233s
* Memory usage for quads: 1803MB
- Finding all 16777216 quads 1048576 times: 31.859s
```

Reply:
The 1:26 to 1:20 is still doing something apparently. but not nearly as much as I'd expect to see in an old V8. Let's keep as-is then!